spacy/strings.pxd

```diff
@@ -7,23 +7,20 @@ from murmurhash.mrmr cimport hash64
 
 from .typedefs cimport attr_t, hash_t
 
-
-cpdef hash_t hash_string(str string) except 0
-cdef hash_t hash_utf8(char* utf8_string, int length) nogil
-
-cdef str decode_Utf8Str(const Utf8Str* string)
-
-
 ctypedef union Utf8Str:
     unsigned char[8] s
     unsigned char* p
 
 
 cdef class StringStore:
     cdef Pool mem
-
     cdef vector[hash_t] keys
-    cdef public PreshMap _map
+    cdef PreshMap key_map
```

Comment:
What do you mean by the name `key_map`? (It sounds like it's opposed to a non-key map or a keyless map?)

Reply:
Don't quite remember why I went for that particular name, but it is indeed confusing. We could call it `values`, given that we store the keys in `keys`. Or we can rename `keys` to `hashes` and `_map` to `hash_to_string_map`.

Reply:
Naming: difficult.

Since it's basically private I don't really mind keeping `_map`.

Actually... do we even really need `keys` at all? Not the functionality, but storing it twice in this `vector`? Is is that much faster than iterating over `_map`?

Reply:
I believe we keep it around to iterate over the strings in insertion order, which isn't guaranteed with the map.

Reply:
Ah, hmm. So in general if you want to add `items()` I think it would make sense to also add `keys()` and `values()`, and then I might rename `keys -> _keys` and then just keep `_map`?

(My other naming ideas are currently all worse.)

Reply:
Sounds good, will do.

Reply:
It just occurred to me that returning the hashes with `keys()` and the strings with `values()` inconsistent with the semantics of `items()` since we return the inverted tuple with the latter.

Since we can't change `__iter__` without breaking code elsewhere, I think that the least confounding option would be to ensure that `keys()`, `values()` and `items()` behave like they would in a `dict`. What do you think?

Reply:
My suggestion was not to return the hashes with `keys()` but to implement `keys()/values()/items()` in a way that's consistent with `__iter__`. This is basically the reason for calling it `_keys`.

We want the python API with `__iter__/keys()/values()/items()` to look like a dict where the strings are the keys and the hashes are the values. We should add some comments explaining how this is backwards from the internal representation/naming, but I think the main thing that matters is that it's consistent for the python API.

Reply:
The types look incorrect for `keys/values/items` and I don't think we want them to be generators. The standard dict things are `MappingView` and I think we want whatever we return to be at least `Iterable` + `Sized`? I don't actually know if we want to use `KeysView`, etc., but I wouldn't expect a generator here.